Detectors/PHOS/calib/src/PHOSL1phaseCalibDevice.cxx

```diff
@@ -41,19 +41,46 @@ void PHOSL1phaseCalibDevice::run(o2::framework::ProcessingContext& pc)
   auto cellTR = pc.inputs().get<gsl::span<TriggerRecord>>("cellTR");
   LOG(detail) << "Processing TF with " << cells.size() << " cells and " << cellTR.size() << " TrigRecords";
   mCalibrator->process(tfcounter, cells, cellTR);
+
+  ++mNprocessed;
+  // If sufficient statistics was collected, try to calculate L1phases
+  // If statistics reached N*thresholds try to re-calculate L1phases
+  // if they changed wrt previous, replace ccdb object, otherwise do nothing
+  if (mNprocessed % minStatisticsForCalib == 0) {
+    minStatisticsForCalib *= 2;
+    mCalibrator->checkSlotsToFinalize(o2::calibration::INFINITE_TF);
+    mCalibrator->endOfStream();
```

Comment:
just to be sure: `mCalibrator->endOfStream()` does not really mean endOfStream, but just to finalize the calibration for the next  portion of TFs, right?

Reply:
Yes, mCalibrator::endOfStream() calculates calibrations but does not reset anything.